src/gpu/generic/sycl/pooling_kernels.hpp

```diff
@@ -124,7 +124,7 @@ struct pooling_fwd_kernel_vec_t {
         return 0;
     }
     float data_conv() const {
-        switch (src_md().data_type()) {
+        switch (dst_md().data_type()) {
```

Comment:
@dzarukin As you mentioned in this [comment](https://github.com/oneapi-src/oneDNN/pull/2039#discussion_r1720244844). It was something else. I think the initialization on the SYCL kernel side was using the src_md data type instead of the dst_md. That was causing the issues. I changed and removed the conditional in the `tests/benchdnn/pool/ref_pool.cpp:L38`. I launched benchdnn tests to be sure nothing else is broken. Let me know what you think and thanks for the feedback.

Reply:
Thanks for double-checking it.